internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -659,3 +664,51 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	xr.SetConditions(xpv1.Available())
 	return reconcile.Result{RequeueAfter: r.pollInterval}, errors.Wrap(r.client.Status().Update(ctx, xr), errUpdateStatus)
 }
+
+// EnqueueForCompositionRevisionFunc returns a function that enqueues (the
+// related) XRs when a new CompositionRevision is created. This speeds up
+// reconciliation of XRs on changes to the Composition by not having to wait for
+// the 60s sync period, but be instant.
+func EnqueueForCompositionRevisionFunc(of resource.CompositeKind, list func(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error, log logging.Logger) func(ctx context.Context, createEvent runtimeevent.CreateEvent, q workqueue.RateLimitingInterface) {
```

Comment:
parameter `of` is used only to obtain `schema.GroupVersionKind` of it - so perhaps we can pass that as the function argument instead, `ofGvk schema.GroupVersionKind`?